.github/workflows/cypress-test.yml

```diff
@@ -0,0 +1,131 @@
+      
+      - name: Create plugins dir
+        run: |
+          cd ./OpenSearch-Dashboards
+          mkdir -p plugins
+      
+      - name: Checkout OpenSearch Dashboard Security plugin
+        uses: actions/checkout@v2
+        with:
+          path: OpenSearch-Dashboards/plugins/security-dashboards-plugin
+          ref: ${{ github.ref }}
+
+      - name: Check OpenSearch Running
+        continue-on-error: true
+        run: curl -XGET https://localhost:9200 -u 'admin:admin' -k
+
+      - name: Get node and yarn versions
+        id: versions
+        run: |
+          echo "::set-output name=node_version::$(cat ./OpenSearch-Dashboards/.node-version)"
+          echo "::set-output name=yarn_version::$(jq -r '.engines.yarn' ./OpenSearch-Dashboards/package.json)"
+      
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ steps.versions.outputs.node_version }}
+          registry-url: 'https://registry.npmjs.org'
+      
+      - name: Install correct yarn version for OpenSearch Dashboards
+        run: |
+          npm uninstall -g yarn
+          echo "Installing yarn ${{ steps.versions_step.outputs.yarn_version }}"
+          npm i -g yarn@${{ steps.versions.outputs.yarn_version }}
+      
+      - name: Check OpenSearch Running
+        continue-on-error: true
+        run: curl -XGET https://localhost:9200 -u 'admin:admin' -k
+
+      - name: Bootstrap OpenSearch Dashboards
+        continue-on-error: false
+        run: |
+          cd ./OpenSearch-Dashboards
+          yarn osd bootstrap
+          echo 'server.host: "0.0.0.0"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.hosts: ["https://localhost:9200"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.ssl.verificationMode: none' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.username: "kibanaserver"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.password: "kibanaserver"' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch.requestHeadersWhitelist: [ authorization,securitytenant ]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.multitenancy.enabled: true' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.multitenancy.tenants.preferred: ["Private", "Global"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.readonly_mode.roles: ["kibana_read_only"]' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.cookie.secure: false' >> ./config/opensearch_dashboards.yml
+          echo 'opensearch_security.multitenancy.enable_aggregation_view: true' >> ./config/opensearch_dashboards.yml
+          yarn start --no-base-path --no-watch &
+          sleep 300
+
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.FTR_PATH }}
+          repository: RyanL1997/opensearch-dashboards-functional-test
```

Comment:
Why do the tests live in another repository?  Seems tedious to make changes in this repository and have to author tests in another repository - how will contributors know what to do?

Reply:
Hi @peternied, thanks for the review. The story behind this is that: before each release, the infra team will build up a [specific build](https://github.com/opensearch-project/opensearch-build/blob/main/src/test_workflow/integ_test/service_opensearch_dashboards.py#L86) with specific configs of all the plugins for only running the functional tests together in functional test repo. As security dashboards plugin, I believe our test should also be included into test set like [this](https://github.com/opensearch-project/opensearch-dashboards-functional-test/blob/main/package.json#L13). According to this, we need to base our tests inside of the functional test repo along with [the other plugins' tests](https://github.com/opensearch-project/opensearch-dashboards-functional-test/tree/main/cypress/integration/plugins).

Reply:
@RyanL1997 How hard it would be to write the tests in this repo and import them into opensearch-dashboards-functional-test?

I understand there is inertia to put tests into that pre-existing repository, consider the risks associated with this dependency because this repo looses more autonomy after we take this change.

#### Potential risk: Unforeseen changes break security-dashboards-plugin
The test repo it can prevent security-dashboards-plugin from being able to merge, while I agree that that this is not likely from a test change the chances this is associated with a dependency that seemed fine to update but breaks this test workflow can create issues.

#### Potential risk: Unblocking security-dashboards-plugin requires merging changes into the dependent codebase 
We might need to get a change in the test repo, since there are many teams contributing there could be a problem in another teams tests that prevent our needed changes from being merged.

#### Potential risk: Merge changes that require simultaneous staging
When making a change in security-dashboards-plugin we could require an update to a test.  If this update does not work without the security-dashboards-plugin change we will be prevented from merging, or have to bypass the checks to merge and hope things pass afterward.

Do we know teams that are using tests for pull request checks?  @opensearch-project/anomaly-detection last year it looks like you adopted out of repo CI tests, have ya'll faced any of the issues above - how has it been?

Reply:
Add @tianleh who helped all dashboards plugin teams onboard the functional test repo. Could you please share your experience when working with plugin teams regarding these potential risks mentioned by Peter? Thanks.

Reply:
Based on my experience with the , I haven't seen the potential risks become real concerns. For the third one about breaking changes, teams may need coordinated merges.

For OSD plugins which are part of the org release, the functional tests will run in infra's Docker env with a specific [Cypress version](https://github.com/opensearch-project/opensearch-build/blob/main/docker/ci/dockerfiles/current/test.rockylinux8.opensearch-dashboards.x64.arm64.dockerfile#L41). Thus keeping the functional tests in the functional test repo makes it easy to manage the Cypress dependency and avoid constraints by the Node version from OSD core.

Reply:
@peternied [Security plugin](https://github.com/DarshitChanpura/opensearch-dashboards-functional-test/tree/main/cypress/integration/plugins/security) and all other plugins are already parts of the functional test repo. Those tests will be added to release pipeline and we are adding a workflow to security-dashboards CI here to make sure that changes in this repo will not introduce regression on this feature.

Reply:
@peternied [Security plugin](https://github.com/DarshitChanpura/opensearch-dashboards-functional-test/tree/main/cypress/integration/plugins/security) and all other plugins are already parts of the functional test repo. Those tests will be added to release pipeline and we are adding a workflow to security-dashboards CI here to make sure that changes in this repo will not introduce regression on this feature.

Reply:
@peternied [Security plugin](https://github.com/DarshitChanpura/opensearch-dashboards-functional-test/tree/main/cypress/integration/plugins/security) and all other plugins are already parts of the functional test repo. Those tests will be added to release pipeline and we are adding a workflow for the Cypress functional tests to security-dashboards-plugin CI here to make sure that changes in this repo will not introduce regressions on this feature.

Reply:
adding @davidlago for awareness or inputs from all maintainers

Reply:
From the anomaly-detection side, we haven't had much concern based on those risks you've mentioned. It has actually been less work in that we don't need to maintain dependencies with Cypress, or needing to upgrade and coordinate with being able to run all Cypress tests from all plugins during bundle builds/tests. We also haven't faced any issues of being blocked from merging changes into the functional-test repo, since it is structured in such a way where plugins can still independently have everything configured for their test suites (the tests, fixtures, etc.).

For the third concern - if there's changes that will break the CI workflow, it is caught in the PR and we usually will create and reference the test change PR to track the updates - that's what we've done before, haven't had issues, since after merging one we will re-run the other to confirm it works (hasn't been to often so not a concern from us).

Reply:
@ohltyler @tianleh Thanks for the input, It sounds like things have been moving smoothly with the test dependency in an outside repo.  While I have concerns we have good counter examples to keep going forward.

Thanks for the discourse @cliu123 and @RyanL1997, I'm resolving this conversation